criu/kerndat.c

```diff
@@ -1099,10 +1119,32 @@ int kerndat_init(void)
 		pr_err("get_ipv6 failed when initializing kerndat.\n");
 		ret = -1;
 	}
-	if (!ret && kerndat_loginuid()) {
-		pr_err("kerndat_loginuid failed when initializing kerndat.\n");
+	if (!ret && kerndat_nsid()) {
+		pr_err("kerndat_nsid failed when initializing kerndat.\n");
 		ret = -1;
 	}
+	if (!opts.uid) {
+		if (!ret && kerndat_loginuid()) {
+			pr_err("kerndat_loginuid failed when initializing kerndat.\n");
+			ret = -1;
+		}
+		if (!ret && kerndat_tun_netns()) {
+			pr_err("kerndat_tun_netns failed when initializing kerndat.\n");
+			ret = -1;
+		}
+		if (!ret && kerndat_socket_unix_file()) {
+			pr_err("kerndat_socket_unix_file failed when initializing kerndat.\n");
+			ret = -1;
+		}
+		if (!ret && kerndat_link_nsid()) {
+			pr_err("kerndat_link_nsid failed when initializing kerndat.\n");
+			ret = -1;
+		}
+		if (!ret && kerndat_socket_netns()) {
+			pr_err("kerndat_socket_netns failed when initializing kerndat.\n");
+			ret = -1;
+		}
```

Comment:
I think it's worth splitting the tests that require root into a helper function

Reply:
Yes, good idea.

Reply:
done